### PR TITLE
Actually use AminoTypes custom type URLs

### PR DESCRIPTION
**Pull request by willclarktech**

### CODE REVIEW

**Review by webmaster128 — COMMENTED** [review]
Could you also test overrides, not only additions? Thanks!

File: packages/stargate/src/aminotypes.ts [hunk 1]
```diff
@@ -27,11 +27,16 @@ export class AminoTypes {
   private readonly register: Record<string, string>;
 
   public constructor(additions: Record<string, string> = {}) {
-    this.register = { ...defaultTypes, ...additions };
+    const additionalAminoTypes = Object.values(additions);
+    const filteredDefaultTypes = Object.entries(defaultTypes).reduce(
+      (acc, [key, value]) => (additionalAminoTypes.includes(value) ? acc : { ...acc, [key]: value }),
+      {},
+    );
```
Comment (webmaster128) [inline packages/stargate/src/aminotypes.ts:34]: Oh, why is this needed?

Comment (willclarktech) [reply packages/stargate/src/aminotypes.ts:34]: Otherwise, it might be many-to-one and `.find` will find the wrong one.

Comment (willclarktech) [reply packages/stargate/src/aminotypes.ts:34]: (I.e. the third test for `fromAmino` will fail.

Comment (webmaster128) [reply packages/stargate/src/aminotypes.ts:34]: Great, thanks

**Review by webmaster128 — APPROVED** [review]
